src/v/redpanda/admin/server.h

```diff
@@ -613,12 +613,24 @@ class admin_server {
     struct level_reset {
         using time_point = ss::timer<>::clock::time_point;
 
-        level_reset(ss::log_level level, time_point expires)
+        level_reset(ss::log_level level, std::optional<time_point> expires)
           : level(level)
           , expires(expires) {}
 
+        friend bool operator<(const level_reset& l, const level_reset& r) {
+            if (!l.expires.has_value() && !r.expires.has_value()) {
+                return false;
+            } else if (!r.expires.has_value()) {
+                return true;
+            } else if (!l.expires.has_value()) {
+                return false;
```

Comment:
nit: could simplify this to just
```suggestion
            if (!l.expires.has_value()) {
                return false;
            } else if (!r.expires.has_value()) {
                return true;
            }
```

Reply:
not a blocker

Reply:
i did another wrong negation in my head. incredible